python/tank/descriptor/io_descriptor/git_tag.py

```diff
@@ -269,39 +189,12 @@ def download_local(self):
         # clear temp file
         filesystem.safe_delete_file(zip_tmp)
```

Comment:
You forgot to delete clone_tmp.

This looks like a pattern that we should implement

with TemporaryFolder() as path:
    do you stuff

regardless of how it leaves the scope, the folder is deleted